src/common/components/elements/CurrentListings.tsx

```diff
@@ -0,0 +1,323 @@
+export const SortByAuctionValues = ['RECENT_BID', 'ENDING_SOONEST'];
+
+const SortByValues = [
+  'MOST_BIDS',
+  'RECENTLY_LISTED',
+  'PRICE_HIGH_TO_LOW',
+  'PRICE_LOW_TO_HIGH',
+  ...SortByAuctionValues,
+] as const;
+export type SortByAction = typeof SortByValues[number]; // keyof typeof sortByFns;
+
+export interface DiscoveryToolState {
+  allListings: Listing[];
+  filteredListings: Listing[];
+  listingsOnDisplay: Listing[];
+  cursor: number;
+  filters: FilterAction[];
+  sortBy: SortByAction; //  SortingActions;
+}
+
+export type FilterOption = { value: FilterAction; label: string };
+const filterOptions: FilterOption[] = [
+  { value: 'SHOW_ALL', label: 'Show all' },
+  { value: 'ACTIVE_AUCTIONS', label: 'Active auctions' },
+  { value: 'BUY_NOW', label: 'Buy now' },
+  { value: 'HAS_1+_BIDS', label: 'Has bids' },
+];
+
+const filterFns: {
+  [fnName in FilterAction]: (l: Listing) => boolean;
+} = {
+  SHOW_ALL: (l) => true,
+  BUY_NOW: (l) => !l.ends_at,
+  ACTIVE_AUCTIONS: (l) => !!l.ends_at,
+  'HAS_1+_BIDS': (l) => !!l.ends_at && !!l.total_uncancelled_bids, // !!l.total_uncancelled_bids is true if it exists and is not 0
+};
+
+const sortByFns: {
+  [fnName in SortByAction]: (a: Listing, b: Listing) => number;
+} = {
+  MOST_BIDS: (a, b) => (b.total_uncancelled_bids || 0) - (a.total_uncancelled_bids || 0),
+  RECENTLY_LISTED: (a, b) => a.created_at.localeCompare(b.created_at),
+  RECENT_BID: (a, b) => {
+    if (!a.last_bid || !b.last_bid) return -1;
+    return a.last_bid - b.last_bid;
+  },
+  PRICE_HIGH_TO_LOW: (a, b) => {
+    const aPrice = a.price_floor || a.instant_sale_price || 0;
+    const bPrice = b.price_floor || b.instant_sale_price || 0;
+    return bPrice - aPrice;
+  },
+  PRICE_LOW_TO_HIGH: (a, b) => {
+    const aPrice = a.price_floor || a.instant_sale_price || 0;
+    const bPrice = b.price_floor || b.instant_sale_price || 0;
+    return aPrice - bPrice;
+  },
+  ENDING_SOONEST: (a, b) =>
+    b.ends_at && a.ends_at ? a.ends_at.localeCompare(b.ends_at) || -1 : -1,
+};
+
+export type SortingOption = { value: SortByAction; label: string };
+const sortingOptions: SortingOption[] = [
+  { value: 'RECENTLY_LISTED', label: 'Recent listings' },
+  { value: 'PRICE_HIGH_TO_LOW', label: 'Expensive' },
+  { value: 'PRICE_LOW_TO_HIGH', label: 'Cheap' },
+  { value: 'RECENT_BID', label: 'Recent bids' },
+  { value: 'MOST_BIDS', label: 'Most bids' },
+  { value: 'ENDING_SOONEST', label: 'Ending soonest' },
+];
+
+export type DiscoveryToolAction =
+  | {
+      type: 'INITIALIZE_LISTINGS';
+      payload: Listing[];
+    }
+  | {
+      type: 'LOAD_MORE_LISTINGS';
+    }
+  | {
+      type: 'FILTER';
+      payload: FilterAction;
+    }
+  | {
+      type: 'SORT';
+      payload: SortByAction;
+    };
+
+const initialState = (options?: {
+  filters?: FilterAction[];
+  sortBy?: SortByAction;
+}): DiscoveryToolState => {
+  const listingShells = Array(8)
+    .fill(null)
+    .map((_, i) => generateListingShell(i.toString()));
```

Comment:
What is the listing shell? Is it something we can drop seems to be putting together object.

Reply:
I'll look to remove it. Used it initially to crate shadow listings, but might not be needed anymore